### DISCUSSION TITLE
Audio optimization

### DISCUSSION THREAD

**Discussion by KentDes** [snakers4/silero-vad#276] [Q&A] [answered]
Hi,

I use voice recognition on my raspberry pi with a dsp (I can configure this dsp (gain, pass filter, ...)).
I want to use your vad with my voice recognition to filter false detection.
Do you have any advice or preferences to optimize your vad (like having min and max db, removing a specific frequency, ...)?
I would think of adding an AGC to improve the result when I am far away but there is more false detection...

Is there an order to implement audio processes (like agc, vad, reveber, noise reduction,...) before your vad?

Thanks

**Comment by snakers4** [comment, marked as answer]
> Do you have any advice or preferences to optimize your vad (like having min and max db, removing a specific frequency, ...)?

All of the knobs are here - https://github.com/snakers4/silero-vad/blob/82d199ff22ae8897946ebb374d216cf6ce4f9aa3/utils_vad.py#L161-L170
Be sure to test on the audio from your domain first, i.e. plot the probability chart and decide on the hyper-params

> I would think of adding an AGC to improve the result when I am far away but there is more false detection...

You can combine our VAD with any other VAD or algorithm and use its features in your algorithm.
The VAD is trained to receive audio as-is without much external interference.

> Is there an order to implement audio processes (like agc, vad, reveber, noise reduction,...) before your vad?

Not sure why adding more noise or reverb before the VAD is helpful, but the VAD just receives audio.

**Reply by KentDes** [reply]
Thank you @Snakers4,

Ok, I'll take a look at the hyper params.

> Not sure why adding more noise or reverb before the VAD is helpful, but the VAD just receives audio.


I was talking about noise cancellation and dereverb (no reverb sorry).
